service/src/test/java/bio/terra/catalog/datarepo/DatarepoClientTest.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.catalog.datarepo;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import bio.terra.catalog.config.DatarepoConfiguration;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.datarepo.client.auth.OAuth;
+import org.junit.jupiter.api.Test;
+
+class DatarepoClientTest {
+
+  private static final String BASE_PATH = "basepath";
+  private static final String TOKEN = "token";
+
+  private final DatarepoClient client;
+
+  DatarepoClientTest() {
+    client = new DatarepoClient(new DatarepoConfiguration(BASE_PATH));
+  }
+
+  @Test
+  void testApis() {
+    var user =
+        new AuthenticatedUserRequest.Builder()
+            .setEmail("")
+            .setSubjectId("")
+            .setToken(TOKEN)
+            .build();
+    var snapshotsClient = client.snapshotsApi(user).getApiClient();
+    assertThat(snapshotsClient.getBasePath(), is(BASE_PATH));
+
+    OAuth oauth = (OAuth) snapshotsClient.getAuthentication("googleoauth");
+    assertNotNull(oauth);
+    assertThat(oauth.getAccessToken(), is(TOKEN));
+
+    var unauthClient = client.unauthenticatedApi().getApiClient();
```

Comment:
should we also assert that the unauth client doesn't have an access token?

Reply:
The bug turned out to be in the test, not in the service code. It's fixed now.